loader/loader.c

```diff
@@ -1050,6 +1050,39 @@ VkResult loader_get_registry_files(const struct loader_instance *inst, char *loc
 
 #endif  // WIN32
 
+void loader_free_layer_properties(const struct loader_instance *inst, struct loader_layer_properties *layer_properties) {
+    if (layer_properties->num_component_layers > 0 && layer_properties->component_layer_names) {
+        loader_instance_heap_free(inst, layer_properties->component_layer_names);
+    }
+    if (layer_properties->num_override_paths > 0 && layer_properties->override_paths) {
+        loader_instance_heap_free(inst, layer_properties->override_paths);
+    }
+    if (layer_properties->num_blacklist_layers > 0 && layer_properties->blacklist_layer_names) {
+        loader_instance_heap_free(inst, layer_properties->blacklist_layer_names);
+    }
+    if (layer_properties->num_app_key_paths > 0 && layer_properties->app_key_paths) {
```

Comment:
If "app_keys" exists but is an empty list then app_key_paths is allocated even though num_app_key_paths == 0. So it should always be freed if it's non-null.

And similar for num_component_layers as well.

Reply:
I had that originally, but noticed some other code which does indeed check the number. Thank you for catching that.